src/azure-cli/azure/cli/command_modules/appconfig/keyvalue.py

```
@@ -283,12 +301,32 @@ def set_key(cmd,
             raise CLIError(str(exception))
 
         if retrieved_kv is None:
+            if content_type and __is_json_content_type(content_type):
+                try:
+                    # Ensure that provided value is valid JSON. Error out if value is invalid JSON.
+                    value = 'null' if value is None else value
+                    json.loads(value)
+                except ValueError:
+                    raise CLIError('Value "{}" is not a valid JSON object, which conflicts with the content type "{}".'.format(value, content_type))
+
             set_kv = KeyValue(key, value, label, tags, content_type)
         else:
+            value = retrieved_kv.value if value is None else value
+            content_type = retrieved_kv.content_type if content_type is None else content_type
+            if content_type and __is_json_content_type(content_type):
+                try:
+                    if value is None:
+                        # retrieved_kv.value must've been null object, which is invalid JSON
```

We can let it fail and will be caught in line 323.